### PR TITLE
delteAll (-a) to delete till end of time

### PR DESCRIPTION
**Pull request by gshatz**
fix bug of deleteAll leaving last partition alive

### CODE REVIEW

File: pkg/tsdb/tsdbtest/tsdbtest.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func SetUp(t testing.TB, v3ioConfig *config.V3ioConfig) func() {
 	return func() {
 		defer metricReporter.Stop()
 		// Don't delete the TSDB table if the test failed
-		if !t.Failed() {
+		if !t.Failed() && !t.Skipped() {
```
Comment (imakhlin) [inline pkg/tsdb/tsdbtest/tsdbtest.go]: Why? `Setup` method should not be invoked at all it if test has been marked as "skipped"
In order to skip table delete attempt when using `deleteAll` you have to provide different Setup/TearDown

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 2]
```diff
@@ -658,18 +661,31 @@ func TestDeleteTable(t *testing.T) {
...
 			data: []tsdbtest.DataPoint{{Time: t1, Value: 222.2},
 				{Time: t2, Value: 333.3},
 				{Time: t3, Value: 444.4}},
 			expected: []tsdbtest.DataPoint{},
 		},
+		{desc: "Should delete all table by deleteAll",
+			deleteFrom:   0,
+			deleteTo:     0,
+			deleteAll:    true,
+			ignoreErrors: true,
```
Comment (imakhlin) [inline pkg/tsdb/v3iotsdb_integration_test.go]: Why all tests pass `ignoreErrors: true`?
IMHO, no errors should be expected during these tests

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 3]
```diff
@@ -713,64 +732,76 @@ func TestDeleteTable(t *testing.T) {
...
-	qry, err := adapter.Querier(nil, 0, time.Now().Unix()*1000)
-	if err != nil {
-		test.Fatalf("Failed to create Querier. reason: %v", err)
-	}
+	if deleteAll {
+		_, err := NewV3ioAdapter(v3ioConfig, nil, nil)
+		if err == nil {
+			test.Fatalf("Table was deleted - failure to create adapter is expected...")
+		}
+		test.Skip("skipping tearDown since table is completely deleted")
```
Comment (imakhlin) [inline pkg/tsdb/v3iotsdb_integration_test.go]: No.
You cannot skip test in the middle. Use proper Setup/TearDown implementation - override default one

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 4]
```diff
@@ -658,18 +661,31 @@ func TestDeleteTable(t *testing.T) {
 		expected     []tsdbtest.DataPoint
 		ignoreReason string
 	}{
-		{desc: "Should delete all table",
+		{desc: "Should delete all table by time",
 			deleteFrom:   0,
 			deleteTo:     9999999999999,
```
[the remark this answers is not in the document]
Comment (gshatz) [reply pkg/tsdb/v3iotsdb_integration_test.go]: no, I'll chnage it to max-int

File: pkg/tsdb/tsdbtest/tsdbtest.go (hunk 1 above)
Comment (gshatz) [reply pkg/tsdb/tsdbtest/tsdbtest.go]: will do it in another pr - must merge this today

File: pkg/tsdb/v3iotsdb_integration_test.go (hunk 3 above)
Comment (gshatz) [reply pkg/tsdb/v3iotsdb_integration_test.go]: will do it tomorrow in another pr

**Pull request merged**